hyperspy/signal.py

```diff
@@ -2482,10 +2482,23 @@ def get_title(signal, i=0):
     def squeeze(self):
         """Remove single-dimensional entries from the shape of an array
         and the axes. See :py:func:`numpy.squeeze` for more details.
+        
+        Returns
+        -------
+        s : signal
+            A new signal object with single-entry dimensions removed
+        
+        Examples
+        --------
+        >>> s = hs.signals.Signal2D(np.random.random((2,1,1,6,8,8)))
+        <Signal2D, title: , dimensions: (6, 1, 1, 2|8, 8)>
+        >>> s = s.squeeze()
+        >>> s
+        <Signal2D, title: , dimensions: (6, 2|8, 8)>
         """
         # We deepcopy everything but data
         self = self._deepcopy_with_new_data(self.data)
-        for axis in self.axes_manager._axes:
+        for axis in reversed(self.axes_manager._axes):
```

Comment:
This will not work all the time, for example:
```python
s = hs.signals.BaseSignal(np.random.random((2,1,1,6,8,8,1,1)))
s2 = s.transpose(signal_axes=[0, 1, 2])
s3 = s2.squeeze()
```
I suspect the navigation and signal axes may need to be iterated separately here.

Reply:
Good catch, a separate iteration over the two seems to do the job. I'll push a commit in a moment.